wolfcrypt/src/port/maxim/max3266x.c

```diff
@@ -0,0 +1,921 @@
+/* max3266x.c
+ *
+ * Copyright (C) 2006-2023 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#warning "Inside max3266x.c"
+#if defined(WOLFSSL_MAX32665) ||  defined(WOLFSSL_MAX32666)
+#warning "Inside max3266x.c define"
+
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+#include <stdint.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <wolfcrypt/src/misc.c>
+#include <wolfssl/wolfcrypt/port/maxim/max3266x.h>
+#include <wolfssl/wolfcrypt/types.h>
+
+#if defined(USE_FAST_MATH) && defined(WOLFSSL_SP_MATH_ALL)
+    #error Cannot Do Both FAST MATH and SP MATH ALL with MAX3266X
+#elif defined(USE_FAST_MATH)
+    #error  MXC Not Compatible with Fast Math 
+    #warning Fast Math is being used
+    #include <wolfssl/wolfcrypt/tfm.h>
+    #define MXC_WORD_SIZE               DIGIT_BIT
+#elif defined(WOLFSSL_SP_MATH_ALL)
+    #warning Assuming SP MATH ALL is being used
+    #include <wolfssl/wolfcrypt/sp_int.h>
+    #define MXC_WORD_SIZE               SP_WORD_SIZE
+#endif
+
+#if defined(MAX3266X_MATH)
+    #if defined(USE_FAST_MATH)
+        #define mxc_mod         fp_mod
```

Comment:
We already have mp_ macros for these... I don't think you need different logic. Just include wolfssl/wolfcrypt/wolf_math.h.